lib/websockets.js

```diff
@@ -3,13 +3,24 @@
 
 var sockets = require.main.require('./src/socket.io/plugins');
 var categoryNotifications = require('./categoryNotifications');
+var controllers = require.main.require("./src/controllers");
 
 sockets.categoryNotifications = {};
 
-sockets.categoryNotifications.subscribe = function(socket, data, callback) {
+sockets.categoryNotifications.subscribe = async function(socket, data, callback) {
 	if (!socket.uid || !data || !data.cid) {
 		return callback(new Error('[[error:invalid-data]]'));
 	}
+
+	try {
+		let isUserBlocked = await controllers.categoryBlockedUsers.isUserBlocked(socket.uid, data.cid);
+		if (isUserBlocked) {
+			return callback(new Error('[[global:blocked_in_category]]'));
+		}
+	} catch (error) {
+		console.log(error);
```

Comment:
don't avoid error like that. return it in callback. So it will block to execute further code.